### PR TITLE
[FIX] base: avoid fail with wrong mimetype

### PR DESCRIPTION
**Pull request by ryce-odoo**
Uploading a WEBP or SVG file disguised with a proper file extension (JPG, PNG) will cause a traceback because img.image is not populated when there is an empty source, SVG, or WEBP file uploaded as this code should not be reached with these file types.

The reason this occurs is because we check for the file extension when deciding to post process an image, but when we get to initializing the ImageProcess object, we then check the actual file structure to verify the type of file.

This is a workaround for the time being, but should not be a final solution in future versions.

Adding a null check on img.image in the _postprocess_contents method in order to avoid attempting to access the size of this image when it is null.

Raises a user error in order to trigger the catch and exit the code while logging the error and 'Post processing ignored:'.

opw-3672250
---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by JKE-be — CHANGES_REQUESTED** [review]
webp and svg are not resizable, so why would you like to add it in ir.config_parameter 'image_autoresize_extensions' ???
If you don't add it, you will just not enter into the `if is_image_resizable and...`'s condition...

Your issue here, is that you try to upload a svg that you hide under the extension jpeg to try to bypass some others rules...
But your commit message is wrong, your UserError too. Maybe we should handle this case to have a better uX, but this commit don't reflect it.

**Comment by ryce-odoo** [comment]
> webp and svg are not resizable, so why would you like to add it in ir.config_parameter 'image_autoresize_extensions' ??? If you don't add it, you will just not enter into the `if is_image_resizable and...`'s condition...
> 
> Your issue here, is that you try to upload a svg that you hide under the extension jpeg to try to bypass some others rules... But your commit message is wrong, your UserError too. Maybe we should handle this case (raise a Warning if we detect wrong mimetype e.g.) to have a better uX, but this commit don't reflect it.

Hey @JKE-be
The issue here is that a customer may not know that they have an SVG or WEBP that had its extension changed. I feel like this is the most efficient way to get around it because raising the error within this try block does not raise any error in the frontend and just exits at the except block saying "Post processing ignored". I feel like this is the best outcome considering some customers may not be tech savvy enough to understand file types.

**Comment by JKE-be** [comment]
so please explain it in the commit msg.

Your problem is not that you really upload a svg file. But that you upload a svg file renamed as .png.
Else you will not enter in this code.

**Comment by ryce-odoo** [comment]
Hey @JKE-be 

Sorry for the late reply was out of office. I have updated the commit message. Please let me know if there is anything else I should change.

**Comment by ryce-odoo** [comment]
Hey @JKE-be 

Yeah it looks like the change that introduced this regression was adding this to the if condition:
"or (source[0:4] == b'RIFF' and source[8:15] == b'WEBPVP8')"

Looks like this change was meant to add compatibility for webp's and with further testing it looks like this issue has always existed with SVG's. However, I don't think anyone would try converting vectors to raster through the file extension like that.

Here's the commit in question:
https://github.com/odoo/odoo/commit/d1292a96a62b18a88a852c69ed7732d16ae0344e

**Comment by bso-odoo** [comment]
@ryce-odoo Can you elaborate on the scenario that causes the error ?
In the opw, it is about an upload in the Documents app. The error also occurs when trying to upload such an image through the website builder inside an image field (e.g. a product image in eCommerce). But other uploads such as an image inside a website page do not cause the error.
Both failing uploads are solved with this PR. :+1: 

This looks good to me. Maybe we should find a way to make the "incoming" mimetype not trustworthy, and have `_compute_mimetype` not rely on the value if it was sent from specific routes, but that might not be a change for a stable version.

**Comment by ryce-odoo** [comment]
Hi @bso-odoo 
I have updated the commit message. Let me know what you think.

**Comment by bso-odoo** [comment]
@ryce-odoo Nice, but what I meant was that it did not mention which upload feature you are talking about.
E.g.: state that, for example, the problem occurs when uploading such a file "in the Documents app".

**Comment by ryce-odoo** [comment]
Hey @bso-odoo 
The issue on the original ticket was from the product documents page, but this happens anywhere where you upload an image to Odoo. It just depends on how much obfuscation is at play if you get this exact error message telling you where the issue is. For example, if you upload it to the chatter or documents app it will just give you a server error and if you upload it to the attachments, product documents, or website you will get the same traceback that tells you what the problem is. I don't know if it is necessary to clarify when this happens because this is an issue with base.

**Comment by ryce-odoo** [comment]
Hey @JKE-be 
Can I get an approval on this PR now? Or is there something else that I should change?

**Comment by ryce-odoo** [comment]
Hey @bso-odoo @JKE-be 

Can either of you guys give this a final check and approve for merge?

File: odoo/addons/base/models/ir_attachment.py
Comment (bso-odoo) [inline odoo/addons/base/models/ir_attachment.py:345]: Sorry I did not notice before, but the error message should a translated text.

For the final approval, I do not have such rights.

**Comment by JKE-be** [comment]
@robodoo delegate=bso-odoo

**Comment by JKE-be** [comment]
@robodoo override=ci/style

**Comment by ryce-odoo** [comment]
Hey @bso-odoo 
I made that changes and added it to the pot file.

File: odoo/addons/base/models/ir_attachment.py
Comment (bso-odoo) [inline odoo/addons/base/models/ir_attachment.py:345]: @ryce-odoo You also need to import `_` for this to work.
Maybe you'll want to add a test to check this scenario too ?

Comment (ryce-odoo) [reply odoo/addons/base/models/ir_attachment.py:345]: @bso-odoo _ is already imported in this file. Not sure if it is worth it to add a test case for a change as small as this as it would involve adding an additional image file to the test folder in base. If you insist I can add it, but it seems excessive.

Comment (bso-odoo) [reply odoo/addons/base/models/ir_attachment.py:345]: @ryce-odoo It is indeed imported at the file level, but I get the following error when I try to upload a WEBP imager enamed to JPG:
```
  File "/home/odoo/src/enterprise/documents/models/ir_attachment.py", line 76, in create
    attachments = super().create(vals_list)
  File "<decorator-gen-59>", line 2, in create
  File "/home/odoo/src/odoo/odoo/api.py", line 415, in _model_create_multi
    return create(self, arg)
  File "/home/odoo/src/odoo/odoo/addons/base/models/ir_attachment.py", line 643, in create
    values = self._check_contents(values)
  File "/home/odoo/src/odoo/odoo/addons/base/models/ir_attachment.py", line 377, in _check_contents
    values = self._postprocess_contents(values)
  File "/home/odoo/src/odoo/odoo/addons/base/models/ir_attachment.py", line 345, in _postprocess_contents
    raise UserError(_("Empty source, SVG, or WEBP"))
TypeError: 'str' object is not callable
```
because `_` is hidden by a string in this scope:
```py
_type, _, _subtype = mimetype.partition('/')
```
Since this small change manages to fail, I believe a test could be useful. :wink:

I guess a test could be simply to create an attachment using the data of any of the existing SVG files in web (or just put one in a string within the test), but forcing the file name and the mimetype to wrong values.

**Comment by ryce-odoo** [comment]
Hey @bso-odoo 
I resolved the scoping issue with _ and included a test. Can you check out the test for me? Should be enough to ensure that the functionality works.

File: odoo/addons/base/tests/test_ir_attachment.py
Comment (bso-odoo) [inline odoo/addons/base/tests/test_ir_attachment.py:265]: Maybe close the tag ? And pretend a bit more this is an SVG ? (This could be any XML) `<svg xmlns="http://www.w3.org/2000/svg"></svg>` ?

The commit title is very vague and does not say what this does, so instead of `[FIX] base: Check img.image for null`, what about something like `[FIX] base: avoid failing when uploading image with wrong extension` ?

**Comment by ryce-odoo** [comment]
Hey @bso-odoo 
I changed the svg to be more legit and changed the commit title. Had to shorten it a bit to fit in the character limit.
Are we good now?

**Comment by bso-odoo** [comment]
@ryce-odoo Thanks for all the changes.

It seems [runbot](https://runbot.odoo.com/runbot/batch/1423405) is not happy :sweat: :
- For the upgrade issue, maybe rebasing your PR on the current saas-16.4 will solve it.
- For the styling [TRY301](https://docs.astral.sh/ruff/rules/raise-within-try/) issue, I guess you'll have to directly log the error message and return - instead of relying on the catch to do it. Maybe you'll have a better idea that would allow for reusing the catch block's text...

**Comment by ryce-odoo** [comment]
Hey @bso-odoo 
I have rebased and made the change to satisfy the style check. Everything seems to be in order now every check has passed.

**Comment by ryce-odoo** [comment]
Hey @bso-odoo 
Sounds good I was confused whether or not to keep the translation, but in the end even with the old version that didn't pass the style check, the message would only ever be logged in the console.

I think its ready to merge when you are.

**Comment by ryce-odoo** [comment]
Thanks @bso-odoo !

**Pull request closed without merging**
